packages/task/src/browser/task-contribution.ts

```diff
@@ -96,38 +96,93 @@ export class TaskResolverRegistry {
      */
     readonly onWillProvideTaskResolver = this.onWillProvideTaskResolverEmitter.event;
 
-    protected resolvers: Map<string, TaskResolver>;
+    protected taskResolvers: Map<string, TaskResolver> = new Map();
+    protected executionResolvers: Map<string, TaskResolver> = new Map();
 
-    @postConstruct()
-    protected init(): void {
-        this.resolvers = new Map();
+    /**
+     * Registers the given {@link TaskResolver} to resolve the `TaskConfiguration` of the specified type.
+     * If there is already a `TaskResolver` registered for the specified type the registration will
+     * be overwritten with the new value.
+     *
+     * @deprecated since 1.12.0 use `registerTaskResolver` instead.
+     *
+     * @param type the task configuration type for which the given resolver should be registered.
+     * @param resolver the task resolver that should be registered.
+     *
+     * @returns a `Disposable` that can be invoked to unregister the given resolver
+     */
+    register(type: string, resolver: TaskResolver): Disposable {
+        return this.registerTaskResolver(type, resolver);
     }
 
     /**
      * Registers the given {@link TaskResolver} to resolve the `TaskConfiguration` of the specified type.
      * If there is already a `TaskResolver` registered for the specified type the registration will
      * be overwritten with the new value.
+     *
      * @param type the task configuration type for which the given resolver should be registered.
      * @param resolver the task resolver that should be registered.
      *
      * @returns a `Disposable` that can be invoked to unregister the given resolver
      */
-    register(type: string, resolver: TaskResolver): Disposable {
-        this.resolvers.set(type, resolver);
+
+    registerTaskResolver(type: string, resolver: TaskResolver): Disposable {
+        this.taskResolvers.set(type, resolver);
         return {
-            dispose: () => this.resolvers.delete(type)
+            dispose: () => this.taskResolvers.delete(type)
         };
     }
 
     /**
      * Retrieves the {@link TaskResolver} registered for the given type task configuration type.
+     *
+     * @deprecated since 1.12.0 use `getTaskResolver()` instead.
+     *
      * @param type the task configuration type
      *
      * @returns a promise of the registered `TaskResolver` or `undefined` if no resolver is registered for the given type.
      */
     async getResolver(type: string): Promise<TaskResolver | undefined> {
+        return this.getTaskResolver(type);
+    }
+
+    /**
+     * Retrieves the {@link TaskResolver} registered for the given type task configuration type.
+     * @param type the task configuration type
+     *
+     * @returns a promise of the registered `TaskResolver` or `undefined` if no resolver is registered for the given type.
+     */
+    async getTaskResolver(type: string): Promise<TaskResolver | undefined> {
```

Comment:
Here I have the same opinion as I mentioned https://github.com/eclipse-theia/theia/pull/9377/files#r620049849:
`getResolver(type: string)` method within ` TaskResolverRegistry` class is readable enough, I believe we shouldn't introduce the new method.

Reply:
And the same response :-)